staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1521,6 +1533,27 @@ func (c *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.No
 			}
 		}
 
+		// IPv6. Ordered after IPv4 addresses, so legacy code can continue to just use the "first" address in a dual-stack cluster.
+		for _, macID := range macIDs {
+			ipPath := path.Join("network/interfaces/macs/", macID, "ipv6s")
+			ips, err := c.metadata.GetMetadata(ipPath)
+			if err != nil {
+				if isAWSNotFound(err) {
```

Comment:
L1487 has other request to metadata but you don't check for the not found error, is this specific of ipv6?